rpy2_rinterface fails to build on Ubuntu 24.04 with system R when libtirpc-dev is missing — no distros/ files declare this dependency

### Steps To Reproduce

1. Fresh Ubuntu 24.04 with `r-base-dev` installed (system R).
2. `libtirpc-dev` not installed.
3. Build Sage from source: `git checkout 10.9.rc2 && make build`
4. Build proceeds for quite some time (hour on my machine), then fails at `rpy2_rinterface-3.6.6` with:

```
/usr/bin/ld: cannot find -ltirpc: No such file or directory
collect2: error: ld returned 1 exit status
```

The link command was:
`/usr/bin/gcc ... -L/usr/lib/R/lib -lR -lpcre2-8 -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n -o /tmp/.../test_pw_r`

R's link flags on Ubuntu 24.04 include `-ltirpc` (R uses libtirpc for RPC functions removed from glibc 2.32+). When `libtirpc-dev` is not installed, the runtime library `libtirpc.so.3` is present but the linker `symlink libtirpc.so` is not, so the link fails.

### Expected Behavior

Either configure should detect the missing `libtirpc-dev` when system R is in use and require it, or `build/pkgs/rpy2_rinterface/distros/debian.txt` should include `libtirpc-dev`. The current behavior fails 80 minutes into the build at the link step, which is an avoidable late failure.

### Actual Behavior

The build fails at the `rpy2_rinterface-3.6.6` link step with:
```
/usr/bin/ld: cannot find -ltirpc: No such file or directory
collect2: error: ld returned 1 exit status
```

The link command was:
`/usr/bin/gcc ... -L/usr/lib/R/lib -lR -lpcre2-8 -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n -o /tmp/.../test_pw_r`

The runtime library `libtirpc.so.3` is present on the system, but the linker symlink `libtirpc.so` (provided by `libtirpc-dev`) is not, so the link fails.

Workaround: `sudo apt install libtirpc-dev`, then `make build` resumes and completes.

### Additional Information

R on Ubuntu 24.04 links against libtirpc because RPC functions were removed from glibc 2.32+. The proximate cause is `R CMD config --ldflags`, which on this system returns:
`-Wl,--export-dynamic -fopenmp -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -Wl,-z,relro -L/usr/lib/R/lib -lR -lpcre2-8 -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n`

rpy2_rinterface's setup.py invokes `R CMD config --ldflags` to obtain link flags for its compile test, so `-ltirpc` is passed to the linker. By contrast, `pkg-config --libs libR` returns only:
`-Wl,--export-dynamic -fopenmp -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -Wl,-z,relro -L/usr/lib/R/lib -lR
`
— without the indirect deps. So this is specific to the R CMD config path.

Verification, with `libtirpc-dev` installed:
```
bash$ echo 'int main(){return 0;}' > /tmp/t.c
$ gcc /tmp/t.c $(R CMD config --ldflags) -o /tmp/t && echo OK || echo FAIL
OK
$ gcc /tmp/t.c $(pkg-config --libs libR) -o /tmp/t && echo OK_pkgconfig || echo FAIL_pkgconfig
OK_pkgconfig
```

Both link paths succeed once `libtirpc-dev` is installed. The pkg-config path also succeeds without libtirpc-dev (only the `R CMD config` path drags in `-ltirpc`), suggesting the tirpc dependency is an over-specification in R's `etc/Makeconf` rather than a real link requirement of libR.

Suggested fix:

`build/pkgs/rpy2_rinterface/` currently has no `distros/` subdirectory at all — the directory contains only `SPKG.rst checksums.ini dependencies package-version.txt spkg-install.in type`. Adding `build/pkgs/rpy2_rinterface/distros/debian.txt` containing `libtirpc-dev` would let `./configure` advise the user before the build begins. (For comparison, `build/pkgs/r/distros/debian.txt` exists and lists R's system dependencies.) Equivalent `fedora.txt` (with `libtirpc-devel`), `arch.txt`, etc. could also be added.

A more robust fix would also add a `spkg-configure.m4` for `rpy2_rinterface` that runs the same compile test (link a trivial program with `R CMD config --ldflags`) and rejects the package at configure time if it fails, guaranteeing the failure is caught early regardless of distro.

### Environment

- Sage version: 10.9.rc2 (also reproducible with earlier 10.9 betas)
- OS: Ubuntu 24.04 on WSL2 (Windows 10 host)
- CPU: x86_64 (Intel i7-5820K)
- System R: r-base-dev from Ubuntu 24.04 packages
- Python: 3.12 (Sage's bundled venv)

### Checklist

- [x] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [x] I have read the documentation and troubleshoot guide

--- a/build/pkgs/r/distros/debian.txt
+++ b/build/pkgs/r/distros/debian.txt
@@ -1,2 +1,3 @@
 r-base
 r-base-dev
+libtirpc-dev

--- a/build/pkgs/r/distros/debian.txt
+++ b/build/pkgs/r/distros/debian.txt
@@ -1,3 +1,3 @@
-r-base-dev
+r-base
 # 29471
 r-cran-lattice

--- a/build/pkgs/r/distros/debian.txt
+++ b/build/pkgs/r/distros/debian.txt
@@ -1,3 +1,2 @@
 r-base
-# 29471
-r-cran-lattice
+r-base-dev

